js/src/collapse.js

```diff
@@ -184,21 +184,21 @@ class Collapse extends BaseComponent {
     this._element.classList.add(CLASS_NAME_COLLAPSING)
     this._element.classList.remove(CLASS_NAME_COLLAPSE, CLASS_NAME_SHOW)
 
-    for (const trigger of this._triggerArray) {
-      const element = getElementFromSelector(trigger)
-
-      if (element && !this._isShown(element)) {
-        this._addAriaAndCollapsedClass([trigger], false)
-      }
-    }
-
     this._isTransitioning = true
 
     const complete = () => {
       this._isTransitioning = false
       this._element.classList.remove(CLASS_NAME_COLLAPSING)
       this._element.classList.add(CLASS_NAME_COLLAPSE)
       EventHandler.trigger(this._element, EVENT_HIDDEN)
```

Comment:
'Hidden' event maybe has to be triggered after transferred code snippet.  
Seems more logical to my eyes, as if we listen a hidden event, we suppose that the classes will be handled already. right?

Reply:
You're totally right, I'm changing it !